site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,995 @@
+
+Since some expressions such as functions can contain other expressions,
+expressions can be represented as a tree. Literal values and field references
+typically are the leaf nodes.
+
+For the expression `INDEX_IN(categories, "Computers") IS NULL`, we have a 
+field reference `categories`, a literal string `"Computers"`, and two functions---
+`INDEX_IN` and `IS NULL`.
+
+![](expression_trees.svg)
+
+The field reference for `categories` is represented by:
+
+```json
+{
+  "selection": {
+    "directReference": {
+      "structField": {
+        "field": 1
+      }
+    },
+    "rootReference": {}
+  }
+}
+```
+
+Whereas SQL references field by names, Substrait always references fields 
+numerically. This means that a Substrait expression only makes sense relative
+to a certain schema. As we'll see later when we discuss relations, for a filter
+relation this will be relative to the input schema, so the `1` here is referring 
+to the second field of `products`.
+
+!!! note
+    Protobuf may not serialize fields with integer type and value 0, since 0 is the default.
+    So if you instead saw `"structField": {}`, know that is is equivalent to 
+    `"structField": { "field": 0 }`.
+
+`"Computers"` will be translated to a literal expression:
+
+```json
+{
+  "literal": {
+    "string": "Computers"
+  }
+}
+```
+
+Both `IS NULL` and `INDEX_IN` will be scalar function expressions. Available 
+functions in Substrait are defined in extension YAML files contained in 
+https://github.com/substrait-io/substrait/tree/main/extensions. Additional
+extensions may be created elsewhere. `IS NULL` is defined as a `is_null` function
+in 
+[functions_comparison.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_comparison.yaml#:~:text=%2D-,name%3A%20%22is_null%22,-description%3A%20Whether)
+and `INDEX_IN` is defined as `index_in` function in
+[functions_set.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_set.yaml#:~:text=%2D-,name%3A%20%22index_in%22,-description%3A%20%3E).
+
+First, the expression for `INDEX_IN("Computers", categories)` is:
+
+<!-- TODO: update once #342 is merged -->
+
+```json
+{
+  "scalarFunction": {
+    "functionReference": 1,
+    "outputType": {
+      "i64": {
+        "nullability": "NULLABILITY_NULLABLE"
+      }
+    },
+    "arguments": [
+      {
+        "enum": {
+          "specified": "NAN_IS_NOT_NAN"
+        }
+      },
+      {
+        "value": {
+          "literal": {
+            "string": "Computers"
+          }
+        }
+      },
+      {
+        "value": {
+          "selection": {
+            "directReference": {
+              "structField": {
+                "field": 1
+              }
+            },
+            "rootReference": {}
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+`functionReference` will be explained later in the plans section. For now,
+understand that it's an index into a list of function definitions and since this
```

Comment:
Let's keep this unless you can come up with something better but this is technically not an "index into a list".  A functionReference of 7 could be the first item in the list or the 10th item.  It's more a key into a map.  However, I can't think of better wording without confusing a reader.

Reply:
I rephrased as an "ID corresponding to an entry in a list" which is hopefully clear and more accurate.